aws/resource_aws_sns_topic_subscription_test.go

```diff
@@ -164,40 +187,164 @@ func TestObfuscateEndpointPassword(t *testing.T) {
 func testAccAWSSNSTopicSubscriptionConfig(i int) string {
 	return fmt.Sprintf(`
 resource "aws_sns_topic" "test_topic" {
-    name = "terraform-test-topic-%d"
+  name = "terraform-test-topic-%d"
 }
 
 resource "aws_sqs_queue" "test_queue" {
-	name = "terraform-subscription-test-queue-%d"
+  name = "terraform-subscription-test-queue-%d"
 }
 
 resource "aws_sns_topic_subscription" "test_subscription" {
-    topic_arn = "${aws_sns_topic.test_topic.arn}"
-    protocol = "sqs"
-    endpoint = "${aws_sqs_queue.test_queue.arn}"
+  topic_arn = "${aws_sns_topic.test_topic.arn}"
+  protocol = "sqs"
+  endpoint = "${aws_sqs_queue.test_queue.arn}"
 }
 `, i, i)
 }
 
 func testAccAWSSNSTopicSubscriptionConfig_filterPolicy(i int, policy string) string {
 	return fmt.Sprintf(`
 resource "aws_sns_topic" "test_topic" {
-    name = "terraform-test-topic-%d"
+  name = "terraform-test-topic-%d"
 }
 
 resource "aws_sqs_queue" "test_queue" {
-	name = "terraform-subscription-test-queue-%d"
+  name = "terraform-subscription-test-queue-%d"
 }
 
 resource "aws_sns_topic_subscription" "test_subscription" {
-    topic_arn = "${aws_sns_topic.test_topic.arn}"
-    protocol = "sqs"
-    endpoint = "${aws_sqs_queue.test_queue.arn}"
-    filter_policy = %s
-  }
+  topic_arn = "${aws_sns_topic.test_topic.arn}"
+  protocol = "sqs"
+  endpoint = "${aws_sqs_queue.test_queue.arn}"
+  filter_policy = %s
+}
 `, i, i, policy)
 }
 
+func testAccAWSSNSTopicSubscriptionConfig_deliveryPolicy(i int, policy string) string {
```

Comment:
Due to security restrictions in the accounts I have access to, I'm unable to run this acceptance test. Please let me know whether you prefer to have this deleted or run it in your end.

Reply:
We generally require acceptance testing for new enhancements and this one does look pretty good overall. It does need to be updated to a more valid delivery policy JSON though, as AWS is currently returning:

```
2018/02/08 11:56:37 [DEBUG] [aws-sdk-go] DEBUG: Validate Response sns/SetSubscriptionAttributes failed, not retrying, error InvalidParameter: Invalid parameter: DeliveryPolicy: healthyRetryPolicy.minDelayTarget must be specified
	status code: 400, request id: 4d4b2c18-4d55-5c2b-af81-7eeeced101fe
```

Reply:
Oh yeah, I forgot the example on the doc is wrong. I'll update this with a valid policy